src/main/plugins/org.dita.html5/xsl/topic.xsl

```diff
@@ -289,6 +284,33 @@ See the accompanying LICENSE file for applicable license.
   <xsl:template match="*[contains(@class, ' topic/titlealts ')]"/>
   
   
+  <!-- =========== HEADING LEVELS =========== -->
+
+  <!-- returns heading level (1 through 6) appropriate at the specified element -->
+  <xsl:function name="dita2html:get-heading-level" as="xs:integer">
+    <xsl:param name="element" as="element()"/>
+    <xsl:sequence select="min((6, count($element/ancestor-or-self::*[dita2html:is-heading-level(.)])))"/>
+  </xsl:function>
+
+  <!-- returns true() for elements that count as a heading level -->
+  <!-- (this is an accessor function to the moded templates below) -->
+  <xsl:function name="dita2html:is-heading-level" as="xs:boolean?">
```

Comment:
This should always return a boolean. You get this by adding a fallback template in `dita2html:is-heading-level` mode that returns `false()`. That also removes the need for `<xsl:mode>` use.